apps/web/composables/useModernImage/useModernImage.ts

```diff
@@ -32,8 +34,50 @@ export const useModernImage: UseModernImageReturn = () => {
     return images.map((image: AgnosticImage) => ({ ...image, url: addModernImageExtension(image.url) }));
   };
 
+  // eslint-disable-next-line unicorn/consistent-function-scoping
+  const getImageForViewport = (product: Product, context: string): string => {
+    if (!product) return '';
+
+    const { isDesktop, isTablet } = useBreakpoints();
+    const isLargeViewport = isDesktop || isTablet;
+
+    switch (context) {
+      case 'ItemList': {
+        return productGetters.getPreviewImage(product);
+      }
+      case 'Whislist': {
+        return isLargeViewport ? productGetters.getFullImage(product) : productGetters.getMiddleImage(product);
+      }
+      case 'CartProductCard': {
+        return isLargeViewport
+          ? productGetters.getSecondPreviewImage(product)
+          : productGetters.getPreviewImage(product);
+      }
+      default: {
+        return '';
+      }
+    }
+  };
+  // eslint-disable-next-line unicorn/consistent-function-scoping
+  const addWebpExtensionForGallerySfImages = (images: ImagesData[]) => {
+    if (useWebp) {
+      return images.map((image: ImagesData) => {
+        return {
+          ...image,
+          url: image.url ? `${image.url}.webp` : '',
+          urlPreview: image.urlPreview ? `${image.urlPreview}.webp` : '',
+          urlMiddle: image.urlMiddle ? `${image.urlMiddle}.webp` : '',
+          urlSecondPreview: image.urlSecondPreview ? `${image.urlSecondPreview}.webp` : '',
+        };
+      });
+    }
+    return images;
+  };
```

Comment:
All this needs a refactor:
- please remember that we now have avif, webp (and we might add more along the way)
- we need to integrate added changes on top of existing ones (from main)
- eslint disabling is a good indicator that we have to improve this part